lib/cartopy/mpl/feature_artist.py

```diff
@@ -114,21 +113,46 @@ def __init__(self, feature, **kwargs):
             color = self._kwargs.pop('color')
             self._kwargs['facecolor'] = self._kwargs['edgecolor'] = color
 
+        # Paths are worked out at draw, but add_collection fails if paths is
+        # left to the default of None.
+        self.set_paths([])
+
         # Set default zorder so that features are drawn under
         # lines e.g. contours but over images and filled patches.
         # Note that the zorder of Patch, PatchCollection and PathCollection
         # are all 1 by default. Assuming default zorder, drawing takes place in
         # the following order: collections, patches, FeatureArtist, lines,
         # text.
-        if self._kwargs.get('zorder') is not None:
-            self.set_zorder(self._kwargs['zorder'])
-        elif feature.kwargs.get('zorder') is not None:
-            self.set_zorder(feature.kwargs['zorder'])
-        else:
-            self.set_zorder(1.5)
+        self.set_zorder(1.5)
+
+        # Update drawing styles from the feature and **kwargs.
+        self.set(**feature.kwargs)
+        self.set(**self._kwargs)
 
         self._feature = feature
 
+    def set_facecolor(self, c):
```

Comment:
Another solution to that might be to have a flag on the feature that says "these are edges" and override `set_color` to only set the edgecolor in that case.  This would make us more consistent with [LineCollection](https://github.com/matplotlib/matplotlib/blob/a6da814873f37a8f8e99ad252e540bbfcde617f9/lib/matplotlib/collections.py#L1452-L1463) which will still allow you to explicitly set facecolor if you really want.  This might also work as a solution for #1782 - though there you probably need to account for some geometries being edges and some not.

Reply:
That does seem like a nice way to go. But, I think what you have here is good and consistent with what we had before, so lets not stall this.

Reply:
Yeah, I was thinking more for a follow-up PR as I'm sure there are pitfalls I haven't thought of!

Thanks for the merge!